Assignment 2.Rmd

```diff
@@ -81,20 +84,96 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
+
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+
+# count of actions received by each student
+EDGES.ST  <- EDGE %>% group_by(to) %>% summarise(sum(count))
+# count of actions give out by each teacher
+EDGES.TE  <- EDGE %>% group_by(from) %>% summarise(sum(count))
+
+# change the column name of data frame, preparing for rows bind
+names(EDGES.ST) <- c("id", "count")
+names(EDGES.TE) <- c("id", "count")
+total_count <- bind_rows(EDGES.ST, EDGES.TE)
+
+# merge this column into VERTEXby their id, and call it as V
+V <- merge(total_count, VERTEX, by="id")
+
+# create the directed graph with its vertex size determined by the action counts
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=V)
+plot(g, layout=layout.fruchterman.reingold, vertex.size=V$count)
+
+```
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+```{r}
+library(tidyr)
+personClass <- read.csv("hudk4050-classes.csv", header=TRUE)
+
+# organize the data to get the columns of name and class titles
+personClass <- personClass %>% unite("name", First.Name:Last.Name, remove=TRUE) %>% 
+  gather(number,title, Class.1:Class.6) %>% 
+  filter(title !="") %>%
+  select(name, title)
+
+# remove all the space between letter and number
+personClass$title <- gsub(' ', '', personClass$title)
+
+# make a person and class matrix with value of 0 and 1
+personClass <- personClass %>% mutate(count = 1) %>% 
+  spread(title, count)
+personClass_matrix <- data.matrix(personClass[,-1])
+rownames(personClass_matrix) <- personClass$name
+personClass_matrix[is.na(personClass_matrix)] <- 0
+
+# remove data that cannot be uninterpreted
+# remove HUDK 4050 since everyone is in this course
+personClass_matrix <- personClass_matrix[,-c(1:4,25)]
+
+# remove the diagnol 
+# otherwise graph is more complex since person will link back to themselves
+personPerson <- personClass_matrix %*% t(personClass_matrix)
+diag(personPerson) <- 0
+
+# make the adjacency graph
+g <- graph.adjacency(personPerson, mode="undirected")
+plot(g, vertex.size = 5, vertex.label.cex = 0.5)
```

Comment:
Your graph looks great 👍 You have done a nice job rendering it so that everything is visible.